samples/order-processing-worker-nestjs/src/activities.provider.ts

```diff
@@ -0,0 +1,122 @@
+import { Future, Result } from "@temporal-contract/boxed";
+import { declareActivitiesHandler, ActivityError } from "@temporal-contract/worker/activity";
+import { orderProcessingContract } from "@temporal-contract/sample-order-processing-contract";
+import { Injectable, Inject } from "@nestjs/common";
+import { ProcessPaymentUseCase } from "./domain/usecases/process-payment.usecase.js";
+import { ReserveInventoryUseCase } from "./domain/usecases/reserve-inventory.usecase.js";
+import { ReleaseInventoryUseCase } from "./domain/usecases/release-inventory.usecase.js";
+import { CreateShipmentUseCase } from "./domain/usecases/create-shipment.usecase.js";
+import { SendNotificationUseCase } from "./domain/usecases/send-notification.usecase.js";
+import { RefundPaymentUseCase } from "./domain/usecases/refund-payment.usecase.js";
+import type { PinoLoggerAdapter } from "./infrastructure/adapters/logger.adapter.js";
+
+/**
+ * Injectable provider that creates type-safe activities with dependency injection
+ *
+ * This demonstrates how to use NestJS DI with Temporal activities:
+ * - Use cases are injected via constructor
+ * - Activities are created using declareActivitiesHandler
+ * - All activities follow the Result/Future pattern for explicit error handling
+ */
+@Injectable()
+export class ActivitiesProvider {
+  constructor(
+    @Inject("LoggerAdapter") private readonly loggerAdapter: PinoLoggerAdapter,
+    private readonly processPaymentUseCase: ProcessPaymentUseCase,
+    private readonly reserveInventoryUseCase: ReserveInventoryUseCase,
+    private readonly releaseInventoryUseCase: ReleaseInventoryUseCase,
+    private readonly createShipmentUseCase: CreateShipmentUseCase,
+    private readonly sendNotificationUseCase: SendNotificationUseCase,
+    private readonly refundPaymentUseCase: RefundPaymentUseCase,
```

Comment:
```suggestion
    @Inject(ProcessPaymentUseCase) private readonly processPaymentUseCase: ProcessPaymentUseCase,
    @Inject(ReserveInventoryUseCase) private readonly reserveInventoryUseCase: ReserveInventoryUseCase,
    @Inject(ReleaseInventoryUseCase) private readonly releaseInventoryUseCase: ReleaseInventoryUseCase,
    @Inject(CreateShipmentUseCase) private readonly createShipmentUseCase: CreateShipmentUseCase,
    @Inject(SendNotificationUseCase) private readonly sendNotificationUseCase: SendNotificationUseCase,
    @Inject(RefundPaymentUseCase) private readonly refundPaymentUseCase: RefundPaymentUseCase,
```

we must use this notation otherwise usecase import could be transformed into type import and it would break